unicorn_contracts/src/contracts_service/updateContractFunction.ts

```diff
@@ -34,103 +26,101 @@ class UpdateContractFunction implements LambdaInterface {
      * @returns {Object} object - API Gateway Lambda Proxy Output Format
      *
      */
-     @tracer.captureLambdaHandler()
-     @metrics.logMetrics()
-     @logger.injectLambdaContext({ logEvent: true })
+    @tracer.captureLambdaHandler()
+    @metrics.logMetrics({ captureColdStartMetric: true })
+    @logger.injectLambdaContext({ logEvent: true })
     public async handler(event: APIGatewayProxyEvent, context: Context): Promise<APIGatewayProxyResult> {
+
+        const modifiedDate = new Date();
+
+        // Assemble from payload
+        let data;
         try {
-            const modifiedDate = new Date();
-
-            // Assemble from payload
-            let data;
-            try {
-                data = await this.validateEvent(event);
-            } catch (error: any) {
-                logger.error(error);
-                const response: APIGatewayProxyResult = {
-                    statusCode: 400,
-                    body: JSON.stringify({ message: (error as Error).message })
-                };
-                tracer.addErrorAsMetadata(error);
-                tracer.addResponseAsMetadata(response, process.env._HANDLER);
-                return response;
-            }
+            data = await this.validateEvent(event);
+        } catch (error) {
+            logger.error('Error during request validation', error as Error);
```

Comment:
Generally speaking, our logger takes two arguments: `message` and extra inputs. In the case of `Error` instances, you pass them directly so that Logger takes care of formatting the stack trace & removing any circular dependencies.